geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/transform/ArrowConversionProcess.scala

```diff
@@ -262,25 +262,28 @@ object ArrowConversionProcess {
         case Some(o) => features.sorted(o).iterator
       }
 
+      buildResult(dictionaries, sorted)
+    }
+
+    private def buildResult(dictionaries: Map[String, ArrowDictionary], sorted: Iterator[SimpleFeature]) = {
       val out = new ByteArrayOutputStream()
-      val writer = SimpleFeatureArrowFileWriter(out, sft, dictionaries, encoding, ipcOpts, sort)
+      // Closing the SimpleFeatureArrowFileWriter should close the ByteArrayOutputStream
+      WithClose(SimpleFeatureArrowFileWriter(out, sft, dictionaries, encoding, ipcOpts, sort)) { writer =>
+        val bytes = ListBuffer.empty[Array[Byte]]
 
-      new Iterator[Array[Byte]] {
-        override def hasNext: Boolean = sorted.hasNext
-        override def next(): Array[Byte] = {
-          out.reset()
+        while (sorted.hasNext) { // send batches
           var i = 0
           while (i < batchSize && sorted.hasNext) {
             writer.add(sorted.next)
             i += 1
           }
-          if (sorted.hasNext) {
-            writer.flush()
-          } else {
-            CloseWithLogging(writer)
-          }
-          out.toByteArray
+          writer.flush()
+          bytes.append(out.toByteArray)
+          out.reset()
         }
+
+        bytes.append(out.toByteArray)
```

Comment:
i think these 2 lines (and the declaration of `bytes`) should be moved outside the WithClose block, so that any footer bytes written by the SimpleFeatureWriter get returned correctly. It might not matter if the footer is missing, but seems safer.

Reply:
Hmm... if ```bytes.append(out.toByteArray)``` is moved outside of the WithClose block, wouldn't ```out``` be closed by the WithClose?